crates/storage/db/src/implementation/mdbx/mod.rs

```diff
@@ -93,15 +102,36 @@ pub struct DatabaseArguments {
 
 impl DatabaseArguments {
     /// Create new database arguments with given client version.
-    pub const fn new(client_version: ClientVersion) -> Self {
+    pub fn new(client_version: ClientVersion) -> Self {
         Self {
             client_version,
+            geometry: Geometry {
+                size: Some(0..(4 * TERABYTE)),
+                growth_step: Some(4 * GIGABYTE as isize),
+                shrink_threshold: Some(0),
+                page_size: Some(PageSize::Set(default_page_size())),
+            },
             log_level: None,
             max_read_transaction_duration: None,
             exclusive: None,
         }
     }
 
+    /// Set the geometry.
+    pub fn with_geometry(
+        mut self,
+        max_size: Option<ByteSize>,
+        growth_step: Option<ByteSize>,
```

Comment:
this should accept the `usize`, so we don't have to import `reth-node-types` just for the `ByteSize`, which in itself is just a wrapper
```suggestion
        max_size: Option<usize>,
        growth_step: Option<usize>,
```

Reply:
Fixed, thanks